src/datapump/jobs/version_update.py

```diff
@@ -0,0 +1,186 @@
+        try:
+            _ = client.create_dataset(self.dataset)
+        except DataApiResponseError:
+            pass
+
+        payload = {
+            "creation_options": {
+                "source_type": "raster",
+                "source_uri": self.source_uri,
+                "source_driver": "GeoTIFF",
+                "data_type": self.data_type,
+                "no_data": self.no_data,
+                "pixel_meaning": self.pixel_meaning,
+                "grid": self.grid,
+                "calc": self.calc
+            },
+            # "metadata": get_metadata(),
+        }
+        _ = client.create_version(self.dataset, self.version, payload)
+
+    def _check_tile_set_status(self) -> JobStatus:
+        client = DataApiClient()
+
+        rts_asset = client.get_asset(self.dataset, self.version, "Raster tile set")
+        if rts_asset["status"] == "saved":
+            return JobStatus.complete
+        elif rts_asset["status"] == "pending":
+            return JobStatus.executing
+        else:
+            return JobStatus.failed
+
+    def _create_tile_cache(self):
+        client = DataApiClient()
+
+        resp = client.get_asset(self.dataset, self.version, "Raster tile set")
+        rts_asset_id = resp["asset_id"]
+
+        payload = {
+            "asset_type": "Raster tile cache",
+            "is_managed": True,
+            "creation_options": {
+                "source_asset_id": rts_asset_id,
+                "min_zoom": 0,
+                "max_zoom": self.max_zoom,
+                "max_static_zoom": 9,
+                "symbology": self.tile_cache_symbology
+            }
+        }
+        _ = client.create_aux_asset(self.dataset, self.version, payload)
+
+    def _check_tile_cache_status(self) -> JobStatus:
+        client = DataApiClient()
+
+        rtc_asset = client.get_asset(self.dataset, self.version, "Raster tile cache")
+        if rtc_asset["status"] == "saved":
+            return JobStatus.complete
+        elif rtc_asset["status"] == "pending":
+            return JobStatus.executing
+        else:
+            return JobStatus.failed
+
+    def _create_aux_assets(self) -> JobStatus:
+        # TODO
+        pass
+
+    def _check_aux_assets_status(self) -> JobStatus:
+        # TODO
+        return JobStatus.complete
+
+    def _mark_latest(self):
+        client = DataApiClient()
+
+        client.set_latest(self.dataset, self.version)
+
+    def _check_latest_status(self) -> JobStatus:
+        client = DataApiClient()
+
+        version_data = client.get_version(self.dataset, self.version)
+
+        if version_data["version"] == self.version:
+            return JobStatus.complete
+        else:
+            return JobStatus.failed
+
+
+# Re-asserting the types of the inherited nested fields is currently necessary
+# because of a limitation of Pydantic
+# see https://github.com/samuelcolvin/pydantic/issues/1296
+class UpdateGLADS2Job(VersionUpdateJob):
```

Comment:
It seems weird to make these subclasses since all you're doing is filling out the parameters. Can you just create a VersionUpdateJob with these parameters? You probably don't even need to hardcode them, you could make them part of your input command and send them as parameters when you call the step function. (this is another reason I'd rather not have external clients call the datapump directly - then they'd need to be responsible for passing the correct parameters for our system).